Document for projecting binding metadata (#36640)
openshift/openshift-docs: applications/connecting_applications_to_services/projecting-binding-data.adoc

[id="projecting-binding-data"]
= Projecting binding data
include::modules/common-attributes.adoc[]
include::modules/servicebinding-document-attributes.adoc[]
:context: projecting-binding-data

toc::[]

This section provides information on how you can consume the binding data.

> == Understanding the consumption of binding data

Reviewer: IBM SG 
`If the subject is a functional overview, begin a heading with words such as Introduction or Overview rather than
contriving a pseudo-task-oriented heading that begins with Understanding, Using, Introducing, or Learning.
Understanding text analytics
Overview of text analytics`

Reviewer: ```suggestion
== Consumption of binding data
```

Author: Thanks, @rolfedh for the suggestion. I have modified the heading.

> For accessing and consuming the binding data from a workload, this data has to be projected into the workload from a backing service. After the backing service exposes the binding data, {servicebinding-title} Operator automatically projects this set of data into the workload in the following methods:

Reviewer: ```suggestion
By default, the {servicebinding-title} Operator projects the binding data from the backing service into the workload by using files. To project the binding data by using environment variables, you configure the `.spec.bindAsFiles` parameter from the `ServiceBinding` resource.
```
This suggestion addresses several issues:

- The way it is used here, "from a workload" can have two different meanings: "For a workload to access and consume..." OR "To access and consume the binding data that comes from a workload..."
- "Automatically" and "manually" are weak words. In most cases, you can improve technical content by removing them and simply describing what the system or user does. 
- When possible, use the simple present tense and active voice instead of other verb forms, such as "projects" instead of "must be projected".
- Help the reader understand what their potential response to the content might be; describe system behavior and user actions separately and  address the reader directly as "you."

Reviewer: Would it help to mention _why_ they might want to use environment variables. 
Is it safe to assume that the user already knows what "binding data" is?

Author: @rolfedh My response:

- I will change it to: "For a workload to access and consume..." 
- Here, the SBO feature automates the projection which is the highlight of the product and the projection feature. The reason why it is an important need/requirement for the user is explained in the Exposing binding data topic (for which I am yet to raise a PR) that will precede this topic. as such I am retaining this "automatically" word. 
- The 1st line is a requirement of the situation. The 2nd statement is the system behavior, not user action.

Author: > Would it help to mention _why_ they might want to use environment variables.
> Is it safe to assume that the user already knows what "binding data" is?

- Users for these procedures are typically Operator providers or users who create backing services. The PR for Exposing binding data will cover this in the Introduction. Given that, the user already is well aware of what binding data is. 

Using SBO, users can choose to project the data as either files (default method) or env variables. these are just options. No particular reason or benefit why a user must choose any one of them. By default, data is mounted by SBO as files.

> For accessing and consuming the binding data from a workload, this data has to be projected into the workload from a backing service. After the backing service exposes the binding data, {servicebinding-title} Operator automatically projects this set of data into the workload in the following methods:

Reviewer: Please rewrite this phrase and the other ones I identify for you using the active voice. In the current draft, the passive voice makes it unclear whether the action you describe is something they must do or is something being done by the system.

> . By default, as files.

Reviewer: ```suggestion
```

Author: As I have said earlier, SBO is offering these two methods from which users can choose and we need to state it explicitly.

> . As environment variables, after you configure the `.spec.bindAsFiles` parameter from the `ServiceBinding` resource.

Reviewer: ```suggestion
```

Author: Same reply as above.

include::modules/sbo-configuration-of-directory-path-to-project-binding-data.adoc[leveloffset=+1]
include::modules/sbo-projecting-the-binding-data.adoc[leveloffset=+1]

== Additional resources
//Commenting out the link to avoid Build errors. The entire SBO section is new with topics. Links will work fine when all the SBO doc PRs will be merged.

> * For more information on how to expose binding data, see Exposing binding data from a service.

Reviewer: ```suggestion
* Exposing binding data from a service
```
For minimalism, you can remove the surrounding text and simply provide the link.

Author: Will implement this.
